### PR TITLE
Add pod kill

### PR DESCRIPTION
**Pull request by haircommander**
With tests, man page, and completions.

Signed-off-by: haircommander <pehunt@redhat.com>

### CODE REVIEW

File: cmd/podman/pod_kill.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	"github.com/projectatomic/libpod/libpod"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+var (
+	podKillFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "all, a",
+			Usage: "Kill processes of containers in all pods",
```
Comment (mheon) [inline cmd/podman/pod_kill.go]: Remove the "processes" - "Kill all containers in all pods"

**Comment by mheon** [comment]
One nit on comments otherwise looks fine.
@rhatdan @TomSweeneyRedHat @umohnani8 PTAL

File: cmd/podman/pod_kill.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"syscall"
+
+	"fmt"
+	"github.com/docker/docker/pkg/signal"
```
Comment (rhatdan) [inline cmd/podman/pod_kill.go]: "fmt" should be with syscall

File: cmd/podman/pod_kill.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+var (
+	podKillFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "all, a",
+			Usage: "Kill all containers in all pods",
+		},
+		cli.StringFlag{
+			Name:  "signal, s",
+			Usage: "Signal to send to the pod",
```
Comment (rhatdan) [inline cmd/podman/pod_kill.go]: If you mention containers above, you should mention them here.

Signall all containers in the pod.

File: cmd/podman/pod_kill.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+			Value: "KILL",
+		},
+		LatestFlag,
+	}
+	podKillDescription = "The main process of each container inside the specified pod will be sent SIGKILL, or any signal specified with option --signal."
+	podKillCommand     = cli.Command{
+		Name:                   "kill",
+		Usage:                  "Kill one or more running pods with a specific signal",
```
Comment (rhatdan) [inline cmd/podman/pod_kill.go]: Send the specified signal or SIGKILL to pods

File: docs/podman-pod-kill.1.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+## DESCRIPTION
+The main process of each container inside the pods specified will be sent SIGKILL, or any signal specified with option --signal.
+
+## OPTIONS
+**--all, -a**
+
+Sends signal to all containers associated with a pod.
+
+**--latest, -l**
```
Comment (TomSweeneyRedHat) [inline docs/podman-pod-kill.1.md]: May have missed it, but do we have the notion of "latest" with pods too?  I know we do with containers.

File: test/e2e/pod_kill_test.go [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+		session = podmanTest.RunTopContainerInPod("", podid2)
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		result := podmanTest.Podman([]string{"pod", "kill", "-a"})
+		result.WaitWithDefaultTimeout()
+		Expect(result.ExitCode()).To(Equal(0))
+		Expect(podmanTest.NumberOfContainersRunning()).To(Equal(0))
+	})
+})
```
Comment (TomSweeneyRedHat) [inline test/e2e/pod_kill_test.go]: Maybe not enough tea yet, but do you have a test with multiple containers in the pod?  Also it doesn't look like you've a tests for when the containers in the pod have already been started.  Might be good to add one or two there.

File: docs/podman-pod-kill.1.md (hunk 5 above)
Comment (haircommander) [reply docs/podman-pod-kill.1.md]: yup I added support last week!

File: test/e2e/pod_kill_test.go (hunk 6 above)
Comment (haircommander) [reply test/e2e/pod_kill_test.go]: RunTopContainerInPod runs an alpine top instance in the pod specified. I can add ones with multiple containers, though.

**Comment by mheon** [comment]
LGTM

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by mheon** [comment]
Going to give it one retry before going to manual merge
@rh-atomic-bot retry

**Pull request closed without merging**